src/main/scala/Ctx.scala

```diff
@@ -1,45 +1,47 @@
 import akka.NotUsed
-import akka.util.Timeout
-import schema.MutationError
 import akka.actor.ActorRef
-import generic.Event
-import generic.MemoryEventStore._
-import generic.View.{Get, GetMany}
 import akka.pattern.ask
 import akka.stream.OverflowStrategy
 import akka.stream.scaladsl.Source
-import org.reactivestreams.Publisher
+import akka.util.Timeout
+import generic.Event
+import generic.PersistentEventStore._
+import generic.View.{Get, GetMany}
+import schema.MutationError
 
+import scala.concurrent.duration._
 import scala.concurrent.{ExecutionContext, Future}
+import scala.language.postfixOps
 
 case class Ctx(
   authors: ActorRef,
   articles: ActorRef,
   eventStore: ActorRef,
-  eventStorePublisher: Publisher[Event],
+  events: Source[Event, NotUsed],
   ec: ExecutionContext,
   to: Timeout
 ) extends Mutation {
   implicit def executionContext = ec
   implicit def timeout = to
 
+  // TODO why buffer?
   lazy val eventStream: Source[Event, NotUsed] =
-    Source.fromPublisher(eventStorePublisher).buffer(100, OverflowStrategy.fail)
+    events.buffer(100, OverflowStrategy.fail).backpressureTimeout(3 seconds)
```

Comment:
To restate what you said to be sure I understood: the purpose of the buffer is that even if the subscriber is not asking for more events, the buffer will. And as a side effect of the bounded buffering, the slow subscriber will get disconnected if the buffer have to keep more than 100 events in memory before the slow subscriber ask for them.

So my added `backpressureTimeout` is a bit useless in that case.
And I can't simply use it to replace the buffer because it won't prevent slowing down the whole stream.

Now I have to see how this interact with the BroadcastHub and if it makes sense to keep the buffer or not :)